elide-core/src/main/java/com/yahoo/elide/graphql/operations/UpdateOperation.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2017, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.graphql.operations;
+
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.HttpStatus;
+import com.yahoo.elide.core.PersistentResource;
+import com.yahoo.elide.core.RequestScope;
+import com.yahoo.elide.graphql.Environment;
+import com.yahoo.elide.graphql.PersistentResourceFetcher;
+import graphql.language.Field;
+import graphql.language.ObjectValue;
+import graphql.language.StringValue;
+import graphql.schema.GraphQLList;
+import graphql.schema.GraphQLObjectType;
+import graphql.schema.GraphQLType;
+
+import javax.ws.rs.WebApplicationException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.yahoo.elide.graphql.ModelBuilder.ARGUMENT_DATA;
+import static com.yahoo.elide.graphql.ModelBuilder.ARGUMENT_ID;
+
+/**
+ * Encapsulate the GraphQL update logic.
+ */
+public class UpdateOperation {
```

Comment:
This class has a different structure than the other operations.  It would be nice if there was a common design pattern here.